deepchecks/core/serialization/common.py

```diff
@@ -277,91 +323,48 @@ def create_failures_dataframe(
     return df
 
 
-def plotlyjs_script(connected: bool = True) -> str:
-    """Return plotly activation script in the requirejs enviroment.
-
-    Parameters
-    ----------
-    connected : bool, default True
-
-    Returns
-    -------
-    str
-    """
-    if connected is True:
-        # Connected so we configure requirejs with the plotly CDN
-        script = textwrap.dedent("""
-            {win_config}
-            {mathjax_config}
-            <script type="text/javascript">
-                if (typeof require !== 'undefined') {{
-                    require.undef("plotly");
-                    requirejs.config({{
-                        paths: {{'plotly': ['{plotly_cdn}']}}
-                    }});
-                    require(
-                        ['plotly'],
-                        function(Plotly) {{
-                            window._Plotly = Plotly;
-                            window.Plotly = Plotly;
-                            console.log('Loaded plotly successfully');
-                        }},
-                        function() {{console.log('Failed to load plotly')}}
-                    );
-                }} else {{
-                    console.log('requirejs is not present');
-                }}
-            </script>
-        """)
-        return script.format(
-            win_config=plotlyhtml._window_plotly_config,
-            mathjax_config=plotlyhtml._mathjax_config,
-            plotly_cdn=plotly_cdn_url().rstrip('.js'),
-        )
-    else:
-        # If not connected then we embed a copy of the plotly.js library
-        script = textwrap.dedent("""
-            {win_config}
-            {mathjax_config}
-            <script type="text/javascript">
-                if (typeof require !== 'undefined') {{
-                    require.undef("plotly");
-                    define('plotly', function(require, exports, module) {{
-                        {script}
-                    }});
-                    require(
-                        ['plotly'],
-                        function(Plotly) {{
-                            window._Plotly = Plotly;
-                            window.Plotly = Plotly;
-                            console.log('Loaded plotly successfully');
-                        }},
-                        function() {{console.log('Failed to load plotly')}}
-                    );
-                }} else {{
-                    console.log('requirejs is not present');
-                }}
-            </script>
-        """)
-        return script.format(
-            script=get_plotlyjs(),
-            win_config=plotlyhtml._window_plotly_config,
-            mathjax_config=plotlyhtml._mathjax_config,
-        )
-
-
-def read_matplot_figures() -> t.List[io.BytesIO]:
+def read_matplot_figures(drawer: t.Optional[t.Callable[[], None]] = None) -> t.List[io.BytesIO]:
```

Comment:
what is drawer?

Reply:
a function that creates matplot figures

`read_matplot_figures` function is created to deal with callable items within the `display` list

Reply:
so add a docstring describing it